content/blog/2022/10/env-var-secrets.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "Scanning secrets in environment variables"
+authors:
+- Raul Cabello Martin
+date: 2022-10-17
+---
+
+We are thrilled to announce you can now scan your environment variables for secrets with the new
+[env-variable-secrets-scanner-policy](https://github.com/kubewarden/env-variable-secrets-scanner-policy)! This policy rejects a `Pod`
+or [workload resources](https://kubernetes.io/docs/concepts/workloads/) such as `Deployments`, `ReplicaSets`, `DaemonSets`
+, `ReplicationControllers`, `Jobs`, `CronJobs` etc. if a secret is found in the environment variable
+within a container, init container, or ephemeral container. Secrets that are leaked in plain text or 
+in base64 encoded variables are detected.
+
+This policy uses [rusty hog](https://github.com/newrelic/rusty-hog), an open source secret scanner from New Relic. The
+policy looks for the following secrets being leaked: RSA private keys, SSH private keys and API tokens for different
+services like Slack, Facebook tokens, AWS, Google, New Relic Keys, etc
+
+This is a perfect example of the real power of `Kubewarden` and `WebAssembly`! We didn't have to write all the complex
+code and regular expressions for scanning secrets. Instead, we used an existing open source library that already
+does this job. We can do this because `Kubewarden` [policies](https://docs.kubewarden.io/writing-policies) are delivered 
+as `WebAssembly` binaries.
+
+Have an idea for a new `Kubewarden` policy? You don't need to write all the code from scratch! You can use 
+your favourite libraries in any of the [supported programming languages](https://docs.kubewarden.io/writing-policies), as long 
+as they can be compiled to `WebAssembly`.
+
+## Let's see it in action!
+
+For this example, a Kubernetes cluster with Kubewarden already installed is required. The installation process is
+described in the [quick start guide](https://docs.kubewarden.io/quick-start).
+
+Let's create a `ClusterAdmissionPolicy` that will scan all pods for secrets in their environment variables:
+
+```
+kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: env-variable-secrets
+spec:
+  module: ghcr.io/kubewarden/policies/env-variable-secrets-scanner-policy:v0.1.0
```

Comment:
```suggestion
  module: ghcr.io/kubewarden/policies/env-variable-secrets-scanner:v0.1.1
```

Reply:
new location of the policy: https://github.com/kubewarden/env-variable-secrets-scanner-policy/pkgs/container/policies%2Fenv-variable-secrets-scanner